project/Build.scala

```diff
@@ -190,7 +190,7 @@ object Build extends sbt.Build {
           "org.mockito" % "mockito-core" % mockitoVersion % "test"
         )
       )
-  ) dependsOn(core % "test->test", streaming % "provided", external_kafka  % "provided")
+  ) dependsOn(streaming % "test->test;compile->compile", streaming % "provided", external_kafka  % "provided")
```

Comment:
Yes, please try sbt assembly packArchive

And check the jar size under target/pack/examples/